core-web/libs/edit-content/src/lib/fields/dot-edit-content-host-folder-field/dot-edit-content-host-folder-field.component.spec.ts

```diff
@@ -0,0 +1,211 @@
+
+import { mockMatchMedia } from '@dotcms/utils-testing';
+
+import { DotEditContentHostFolderFieldComponent } from './dot-edit-content-host-folder-field.component';
+
+import { DotEditContentService } from '../../services/dot-edit-content.service';
+import {
+    HOST_FOLDER_TEXT_MOCK,
+    TREE_SELECT_MOCK_NODE,
+    TREE_SELECT_SITES_MOCK,
+    TREE_SELECT_MOCK,
+    createFormGroupDirectiveMock
+} from '../../utils/mocks';
+
+describe('DotEditContentHostFolderFieldComponent', () => {
+    let spectator: Spectator<DotEditContentHostFolderFieldComponent>;
+    let component: DotEditContentHostFolderFieldComponent;
+    let service: SpyObject<DotEditContentService>;
+
+    const createComponent = createComponentFactory({
+        component: DotEditContentHostFolderFieldComponent,
+        componentViewProviders: [
+            { provide: ControlContainer, useValue: createFormGroupDirectiveMock() }
+        ],
+        providers: [
+            FormGroupDirective,
+            mockProvider(DotEditContentService, {
+                getSitesTreePath: jest.fn().mockReturnValue(of(TREE_SELECT_SITES_MOCK))
+            })
+        ],
+        detectChanges: false
+    });
+
+    beforeEach(() => {
+        spectator = createComponent({
+            props: {
+                field: {
+                    ...HOST_FOLDER_TEXT_MOCK
+                }
+            }
+        });
+        service = spectator.inject(DotEditContentService);
+        component = spectator.component;
+        component.formControl.setValue(null);
+        mockMatchMedia();
+    });
+
+    it('should create the component', () => {
+        spectator.detectChanges();
+        expect(spectator.component).toBeTruthy();
+    });
+
+    it('should show options', () => {
+        spectator.detectChanges();
+        const options = component.options();
+
+        expect(service.getSitesTreePath).toHaveBeenCalled();
+        expect(options).toBe(TREE_SELECT_SITES_MOCK);
+        expect(component.treeSelect.options).toBe(TREE_SELECT_SITES_MOCK);
+    });
+
+    describe('The init value with the root path', () => {
+        it('should show a root path', () => {
+            const rootPath = TREE_SELECT_SITES_MOCK[0].key;
+            component.formControl.setValue(rootPath);
+            spectator.detectChanges();
+            const label = spectator.query('.p-treeselect-label');
+            expect(label).toHaveText(rootPath);
+        });
+
+        it('should show the selected path in the tree', () => {
+            const rootPath = TREE_SELECT_SITES_MOCK[0].key;
+            component.formControl.setValue(rootPath);
+            spectator.detectChanges();
+
+            const triggerElement = spectator.query('.p-treeselect-trigger');
+            spectator.click(triggerElement);
+            const labelElement = spectator.query('.p-treenode-content.p-highlight');
+            expect(labelElement).toHaveText(rootPath);
```

Comment:
Here you are technically testing the internal funcionality of a 3rd party component, can we just check on our side that we are passing the right option "selected"?

Reply:
@fmontes Yes, I understand that we should not use 3rd party, but in this case, I would like to test that when a select the component init with a path with levels should be selected by default and applied the pipe, if only ask for selectNode in TreeSelect component do not have this result.